sw/device/silicon_creator/rom/second_rom_start.S

```diff
@@ -0,0 +1,321 @@
+  .endr
+
+  // Entry 31: non-maskable interrupt handler.
+  j rom_nmi_handler
+  .size _rom_interrupt_vector_c, .-_rom_interrupt_vector_c
+
+  // Pop ROM interrupt vector options.
+  //
+  // Re-enable compressed instructions, linker relaxation.
+  .option pop
+
+  /**
+   * ROM shadow stack.
+   */
+  .section .bss
+  .balignl 4, UNIMP
+  .global _rom_shadow_stack
+  .type _rom_shadow_stack, @object
+_rom_shadow_stack:
+  .zero 256 * 4
+  .size _rom_shadow_stack, .-_rom_shadow_stack
+
+// -----------------------------------------------------------------------------
+
+/**
+ * ROM runtime initialization code.
+ */
+
+  // NOTE: The "ax" flag below is necessary to ensure that this section
+  // is allocated executable space in ROM by the linker.
+  .section .crt, "ax"
+
+  // Linker relaxations are disabled until `gp` is set below, because otherwise
+  // some sequences may be turned into `gp`-relative sequences, which is
+  // incorrect when `gp` is not initialized.
+  .option push
+  .option norelax
+
+  /**
+   * Entry point after base ROM.
+   */
+  .balignl 4, UNIMP
+  .global _second_rom_start_boot
+  .type _second_rom_start_boot, @function
+_second_rom_start_boot:
+  // Set up the global pointer and re-enable linker relaxations.
+  la gp, __global_pointer$
+  .option pop
+
+LABEL_FOR_TEST(kRomStartBootMaybeHalt)
+  // Check if we should halt here.
+  li   a0, (TOP_DARJEELING_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_ROM_EXEC_EN_OFFSET(a0)
+  bnez t0, .L_rma_spin_check_straps
+LABEL_FOR_TEST(kRomStartBootHalted)
+.L_halt_loop:
+  wfi
+  j .L_halt_loop
+
+.L_rma_spin_check_straps:
+  // Configure strap pins.
+  //
+  // | GPIO Pin:        | 24             | 23             | 22               |
+  // |                  | (SW_STRAP_2)   | (SW_STRAP_1)   | (SW_STRAP_0)     |
+  // |------------------+----------------+----------------+------------------|
+  // | RMA Entry Value: | Strong pull-up | Strong pull-up | Strong pull-down |
+  // |------------------+----------------+----------------+------------------|
+  // | Configuration:   | Input          | Input          | Input            |
+  // |                  | Pull-down      | Pull-down      | Pull-up          |
+  li   a0, (TOP_DARJEELING_PINMUX_AON_BASE_ADDR + \
+            PINMUX_MIO_PAD_ATTR_0_REG_OFFSET)
+  li   t0, (1 << PINMUX_MIO_PAD_ATTR_0_PULL_EN_0_BIT) | \
+           (1 << PINMUX_MIO_PAD_ATTR_0_PULL_SELECT_0_BIT)
+  sw   t0, (SW_STRAP_0_PAD * 4)(a0)
+  li   t0, (1 << PINMUX_MIO_PAD_ATTR_0_PULL_EN_0_BIT)
+  sw   t0, (SW_STRAP_1_PAD * 4)(a0)
+  sw   t0, (SW_STRAP_2_PAD * 4)(a0)
+  li   a0, (TOP_DARJEELING_PINMUX_AON_BASE_ADDR + \
+            PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET)
+
+  // Spin for a minimum of 5us worth of cycles (assuming 100MHz clock), while we
+  // wait for the RMA strap pull enables to propagate to the physical pads.
+  // Since this is a wait loop, disable the watchdog.
```

Comment:
Same here, I don't see the watchdog actually being disabled.

Reply:
Fixed as well, thanks.